test/GithubApi.Repositories.test.js

```diff
@@ -0,0 +1,81 @@
+const agent = require('superagent');
+const chai = require('chai');
+const md5 = require('md5');
+
+chai.use(require('chai-subset'));
+
+const { expect } = chai;
+
+const url = 'https://api.github.com';
+
+describe('Given a user the github', () => {
+  const username = 'aperdomob';
+
+  describe('when get user', () => {
+    let repositories;
+    let repository;
+    let user;
+    const repositorySearch = 'jasmine-awesome-report';
+
+    before(async () => {
+      const userResponse = await agent.get(`${url}/users/${username}`).auth('token', process.env.ACCESS_TOKEN).set('User-Agent', 'agent');
+      user = userResponse.body;
+      const repositoriesResponse = await agent.get(`${url}/users/${username}/repos`).auth('token', process.env.ACCESS_TOKEN).set('User-Agent', 'agent');
+      repositories = repositoriesResponse.body;
+      repository = repositories.find((rep) => rep.name === repositorySearch);
+    });
+
+    it('Then the user should have been obtained', () => {
+      expect(user.name).to.equal('Alejandro Perdomo');
+      expect(user.company).to.equal('PSL');
+      expect(user.location).to.equal('Colombia');
+    });
+
+    it('Then should have the repository', () => {
+      expect(repository.full_name).to.equal('aperdomob/jasmine-awesome-report');
+      expect(repository.private).to.equal(false);
+      expect(repository.description).to.equal('An awesome html report for Jasmine');
+    });
+
+    describe('When download the zip of repository', () => {
+      let zip;
+
+      before(async () => {
+        const downloadResponse = await agent.get(`${repository.svn_url}/archive/master.zip`).auth('token', process.env.ACCESS_TOKEN).set('User-Agent', 'agent').buffer(true);
+        zip = downloadResponse;
```

Comment:
I downloaded it with a different url that was a property of the response of the repository. Maybe this one also works (supongo...) but try to use hypermedia.